### PR TITLE
[cspell] migrate custom words to upstream dictionaries

### PR DESCRIPTION
**Pull request by hereisnaman**
Once merged this will remove the custom words mentioned in the cspell config. Those words are under migration to the appropriate cspell dictionaries.

**fixes:** #2298 

**dependency:** https://github.com/streetsidesoftware/cspell-dicts/pull/184 (DO NOT MERGE BEFORE THIS)

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "language": "en",
+  "dictionaries": ["softwareTerms"],
```
Comment (IvanGoncharov) [inline cspell.json:3]: If `softwareTerms` allows removing some words can you please do this in separate PR?

**Comment by IvanGoncharov** [comment]
@hereisnaman Also didn't mention in original issue but can you contribute these words into JS-specific vocabularies?
```
    "jsutils",
    "tsutils",
    "noflow",
    "flowlint",
```

File: cspell.json (hunk 1 above)
Comment (hereisnaman) [reply cspell.json:3]: Most of our custom words have been added to the `softwareTerms` dictionary in the mentioned PR. And changes to both `en_us` and `software_terms` at `cspell_dicts` will be merged in a single PR only.

Comment (hereisnaman) [reply cspell.json]: Also, just re-checked, all the dicts are auto included.

**Comment by hereisnaman** [comment]
> @hereisnaman Also didn't mention in original issue but can you contribute these words into JS-specific vocabularies?
> 
> ```
>     "jsutils",
>     "tsutils",
>     "noflow",
>     "flowlint",
> ```

Updated both PRs.
